src/common/playlist/playlist-item.js

```diff
@@ -2,38 +2,41 @@
 const formats = ['hls', 'dash', 'progressive'];
 /**
  * @class PlaylistItem
- * @param {ProviderMediaConfigSourcesObject} [sources] - The item sources
+ * @param {PKSourcesConfigObject} [sources] - The item sources
  * @param {KPPlaylistItemConfigObject} [config] - The item config
  */
 class PlaylistItem {
-  _sources: ?ProviderMediaConfigSourcesObject;
+  _sources: ?PKSourcesConfigObject;
   _config: ?KPPlaylistItemConfigObject;
 
-  constructor(sources: ?ProviderMediaConfigSourcesObject, config: ?KPPlaylistItemConfigObject) {
+  constructor(sources: ?PKSourcesConfigObject, config: ?KPPlaylistItemConfigObject) {
     this._sources = sources;
     this._config = config;
   }
 
   /**
    * Update the playlist item sources
-   * @param {ProviderMediaConfigSourcesObject} sourcesObject - The sources
+   * @param {PKSourcesConfigObject} sourcesObject - The sources
    * @returns {void}
    * @instance
    * @memberof PlaylistItem
    */
-  updateSources(sourcesObject: ProviderMediaConfigSourcesObject): void {
+  updateSources(sourcesObject: PKSourcesConfigObject): void {
     formats.forEach(format => {
       this._sources && (this._sources[format] = sourcesObject[format]);
     });
+    if (this._sources && sourcesObject.options) {
```

Comment:
doesn't it enough to replace the source object by `sourcesObject `?

Reply:
Since now I'm taking the sources from the config (not from `loadMedia` response which can be overridden by app config) you are right.